app/src/main/java/de/xikolo/presenters/main/CertificateListPresenter.java

```diff
@@ -0,0 +1,100 @@
+package de.xikolo.presenters.main;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import de.xikolo.managers.CourseManager;
+import de.xikolo.managers.UserManager;
+import de.xikolo.models.Course;
+import de.xikolo.models.Enrollment;
+import de.xikolo.presenters.base.LoadingStatePresenter;
+import io.realm.Realm;
+import io.realm.RealmResults;
+
+public class CertificateListPresenter extends LoadingStatePresenter<CertificateListView> {
+
+    private CourseManager courseManager;
+
+    private Realm realm;
+
+    private List<Course> courseList;
+
+    private RealmResults coursePromise;
+
+    CertificateListPresenter() {
+        this.courseManager = new CourseManager();
+        this.realm = Realm.getDefaultInstance();
+        this.courseList = new ArrayList<>();
+    }
+
+    @Override
+    public void onViewAttached(CertificateListView view) {
+        super.onViewAttached(view);
+
+        if (courseList == null || courseList.size() == 0) {
+            requestCourses(false);
+        }
+
+        update();
+    }
+
+    @Override
+    public void onViewDetached() {
+        super.onViewDetached();
+
+        if (coursePromise != null) {
+            coursePromise.removeAllChangeListeners();
+        }
+    }
+
+    private void update() {
+        coursePromise = courseManager.listCourses(realm, courses -> {
+            courseList = realm.copyFromRealm(courses);
+            for (int i = 0; i < courseList.size(); i++) {
+                Enrollment e = Enrollment.getForCourse(courseList.get(i).id);
+                if (e == null
+                    || e.certificates == null
+                    || (e.certificates.confirmationOfParticipationUrl == null
+                    && e.certificates.recordOfAchievementUrl == null
+                    && e.certificates.qualifiedCertificateUrl == null)) {
```

Comment:
Looks like a good model method for enrollment, something like `fun anyCertificateAchieved(): Boolean`.